scipy/signal/tests/test_spectral.py

```diff
@@ -1092,14 +1093,21 @@ def test_normalize(self):
         pgram = lombscargle(t, x, f)
         pgram2 = lombscargle(t, x, f, normalize=True)
 
+        # Calculate the scale to convert from unnormalized to normalized
+        weights = np.ones_like(t)/float(t.shape[0])
+        Y_sum = (weights * x).sum()
+        YY_hat = (weights * x * x).sum()
+        YY = YY_hat - Y_sum * Y_sum
+        scale_to_use = 2/(YY*t.shape[0])
+
         # check if normalization works as expected
-        assert_allclose(pgram * 2 / np.dot(x, x), pgram2)
+        assert_allclose(pgram * scale_to_use, pgram2)
         assert_approx_equal(np.max(pgram2), 1.0, significant=2)
 
     def test_wrong_shape(self):
         t = np.linspace(0, 1, 1)
         x = np.linspace(0, 1, 2)
-        f = np.linspace(0, 1, 3)
+        f = np.linspace(0, 1, 3) + 0.1  # will raise 2 errors if f contains 0
```

Comment:
Two errors were being raised here, but the f=0 error is not what was being tested for.